cypress_test/integration_tests/common/helper.js

```diff
@@ -1031,6 +1031,39 @@ function textSelectionShouldNotExist() {
 	cy.log('Make sure there is no text selection - end.');
 }
 
+// Used to represent the bounds of overlays like cell-cursor, document selections etc.
+class Bounds {
+	constructor(boundsJsonString) {
+		var jsonObject = JSON.parse(boundsJsonString);
+		this.top = jsonObject.top;
+		this.left = jsonObject.left;
+		this.width = jsonObject.width;
+		this.height = jsonObject.height;
+	}
+}
+
+// This ensures that the overlay item of the given name has the expected bounds.
+// Parameters:
+// itemName - The name of the overlay item that helps identify the
+//            corresponding test div element.
+// boundsString - A valid JSON string representing the expected bounds of
+//                the overlay item. It must have the keys: top, left, width and height
+//                for a valid bound.
+function overlayItemHasBounds(itemName, boundsString) {
+	cy.log('The overlay item = ' + itemName + ' is expected to have the bounds : ' + boundsString);
+
+	var itemDivId = '#test-div-overlay-' + itemName;
+	return cy.get(itemDivId)
+		.should('exist')
+		.then(function (elem) {
```

Comment:
You should use `.should()` here instead of `.then()`. The difference is that `then` is called only once and so the assertion is checked once, while with `should` the assertion is checked until it passes or reaches a timeout. That latter one is what we generally use because the selection overlay for example is not updated instantly, so with `then` the assertion can fail because it runs before the selection overlay is updated. However, with `should` selection overlay has time to be updated. The test case you commented out, might be fixed by this change.